tests/mock_vllm_api_server.py

```diff
@@ -13,8 +13,14 @@
 
 # import classes to mock
 from vllm.worker.tt_worker import TTWorker, TTCacheEngine
-from mock_vllm_model import new_init_cache_enginer, new_allocate_kv_cache, MockModel
+from mock_vllm_model import (
+    new_init_cache_enginer,
+    new_allocate_kv_cache,
+    MockModel,
+    init_wrapper,
+)
 from vllm.engine.multiprocessing.engine import run_mp_engine
+from vllm.engine.llm_engine import LLMEngine
```

Comment:
Any reason why it's `LLMEngine` and not `AsyncLLMEngine` for the online server?

Reply:
I believe `AsyncLLMEngine` is a wrapper for `LLMEngine`. For the online server, the logging functionality still gets implemented in an `LLMEngine` . But for this case, [line 127 always returns false](https://github.com/tenstorrent/vllm/blob/dev/vllm/entrypoints/openai/api_server.py#L127C27-L127C48)  (since `pipeline_parellel_size = 1` so  [`is_unsupported_config()`] (https://github.com/tenstorrent/vllm/blob/main/vllm/engine/multiprocessing/client.py#L120) returns false. This means `MQLLMEngine` is created. The `LLMEngine` object is created in [`MQLLMEngine` ](https://github.com/tenstorrent/vllm/blob/dev/vllm/engine/multiprocessing/engine.py#L78).  Since the relevant attributes are created in `LLMEngine` that is why it is used as opposed to `MQLLMEngine`

Reply:
Got it -- thank you for the explanation.